internal/metrics/metrics.go

```diff
@@ -220,8 +222,15 @@ func ReportCrossZoneMetric(ctx context.Context, driverNodeName string, executorN
 	totalPairs := int64(totalNumPods * (totalNumPods - 1) / 2)
 	numberOfZones := int64(len(numPodsPerZone))
 
-	metrics.FromContext(ctx).Histogram(crossAzTraffic).Update(crossZonePairs)
-	metrics.FromContext(ctx).Histogram(totalTraffic).Update(totalPairs)
+	crossAzTraffic := metrics.FromContext(ctx).Histogram(crossAzTraffic)
+	crossAzTraffic.Update(crossZonePairs)
+	totalTraffic := metrics.FromContext(ctx).Histogram(totalTraffic)
+	totalTraffic.Update(totalPairs)
+
+	// Needed because the mean is stripped from the metric logs of histograms by default, we need to explicitly update it
+	metrics.FromContext(ctx).GaugeFloat64(crossAzTrafficMean).Update(crossAzTraffic.Mean())
+	metrics.FromContext(ctx).GaugeFloat64(totalTrafficMean).Update(totalTraffic.Mean())
+
 	metrics.FromContext(ctx).Histogram(applicationZonesCount).Update(numberOfZones)
```

Comment:
I'm really not a fan of this but it looks like there's not much we can do here, happy to discuss

Reply:
Haven't checked the `crossAzTraffic` calculation, but maybe we should discuss why the mean is what we care about in this instance